fenix/docs/architecture-overview.md

```diff
@@ -2,168 +2,177 @@
 
-#### Description
-This is where much of the business logic of the app lives. Whenever called by an Interactor, a Controller will do one of the three following things:
-- Create a new [Action](#action) that describes the necessary change, and send it to the Store
-- Navigate to a new fragment via the NavController. Optionally include any state necessary to create this new fragment
-- Interact with some third party manager. Typically these will update their own internal state and then emit changes to an observer, which will be used to update our Store
-
-Controllers can become very complex, and should be unit tested thoroughly whenever their methods do more than delegate simple calls to other objects.
+The Store will create a chain of Middleware instances and invoke them in order. Every Middleware can decide to continue the chain or disrupt the chain. A Middleware has no knowledge of what comes before or after it in the chain.
 
 -------
 
-### <a name="view"/>View
-#### Overview
+### <a name="view">View</a>
+#### **Overview**
 Initializes UI elements, then updates them in response to [State](#state) changes
 
 Observes: [Store](#store)
 
-Calls: [Interactor](#interactor)
+#### **Description**
+The view defines the mapping of State to UI. This can include XML bindings, Composables, or anything in between.
 
-#### Description
-The view defines the mapping of State to UI. This includes initial setup of Views, and also typically includes an `update(state)` function that is called whenever State is changed.
+Views should be as dumb as possible, and should include little or no conditional logic outside of determining which branch of a view tree to display based on State. Ideally, each primitive value in a State object is set on some field of a UI element.
 
-Views should be as dumb as possible, and should include little or no conditional logic. Ideally, each primitive value in a State object is set on some field of a UI element, with no other logic included.
+Views set listeners to UI elements, which trigger dispatches of [Actions](#action) to [Stores](#store).
 
-Views set listeners on to UI elements, which trigger calls to one or more Interactors.
+In some cases, it can be appropriate to initiate side-effects from the view when observing State updates from the Store. For example, a menu might be displayed.
 
 -------
 
 ## Important notes
-- Unlike other common implementations of unidirectional data flow, which typically have one global Store of data, we maintain a smaller Store for each screen.
+- Unlike other common implementations of unidirectional data flow, which typically have one global Store of data, we maintain smaller Stores for each screen and several global Stores.
+  - This is largely to operate within the physical hardware constraints presented by Android development, like having more limited memory resources.
```

Comment:
I think the reasoning is also that there is no need to maintain a UI state when the UI itself is destroyed